pkgs/by-name/do/dokieli/package.nix

```diff
@@ -2,34 +2,33 @@
   lib,
   stdenv,
   fetchFromGitHub,
-  fetchYarnDeps,
-  yarnConfigHook,
   makeWrapper,
-  yarnBuildHook,
-  yarnInstallHook,
   nodejs,
   xsel,
+  yarn-berry_4,
+  unstableGitUpdater,
 }:
 
+let
+  yarn-berry = yarn-berry_4;
+in
+
 stdenv.mkDerivation (finalAttrs: {
   pname = "dokieli";
-  version = "0-unstable-2024-12-12";
+  version = "0-unstable-2025-04-28";
 
-  # Can't update newer versions currently because newer versions require yarn-berry, and it's not in nixpkgs, yet.
   src = fetchFromGitHub {
     owner = "linkeddata";
     repo = "dokieli";
-    rev = "d8dc72c81b84ec12f791892a6377a7f6ec46ed3b";
-    hash = "sha256-CzSyQVyeJVOP8NCsa7ST3atG87V1KPSBzTRi0brMFYw=";
+    rev = "eda3da244e7224ef3819be47e9bf101aceee2701";
+    hash = "sha256-jB3xomCeKfU/Dbl5s2QPLwkUiifKWRcQhgokoFwB7Ws=";
   };
 
-  offlineCache = fetchYarnDeps {
-    yarnLock = "${finalAttrs.src}/yarn.lock";
-    hash =
-      if stdenv.hostPlatform.isDarwin then
-        "sha256-bw5HszcHZ60qgYgm4qfhZEYXjJAQ2DXhWU0Reqb9VpQ="
-      else
-        "sha256-rwHBDBWZe4cdTyL7lNkB4nlpd5MWzbTU6kzdLBWcq0M=";
+  passthru.updateScript = unstableGitUpdater { };
+
+  offlineCache = yarn-berry.fetchYarnBerryDeps {
+    inherit (finalAttrs) src;
+    hash = "sha256-ROGLDrLrvFRbnIDSrrklMYUa1TCf/cfgucQzG7FnSBc=";
   };
 
   installPhase = ''
```

Comment:
Please add the `preInstall` and `postInstall` hooks ([see docs](https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/stdenv.chapter.md#phases-sec-stdenv-phases))